### PR TITLE
Update data to map

### PR DESCRIPTION
**Pull request by rohit2sharma95**
Ref: https://github.com/twbs/bootstrap/issues/32123#issuecomment-730180864

Though, this PR is missing some unit tests for data.js 😄

### CODE REVIEW

File: js/src/dom/data.js [hunk 1]
```diff
@@ -12,55 +12,29 @@
  */
 
 const mapData = (() => {
```
Comment (alpadev) [inline js/src/dom/data.js]: I think you can use the `Map` here directly without the iife

Comment (rohit2sharma95) [reply js/src/dom/data.js]: True, IIFE was to keep the inner variables private before, but I am not sure if still, we need to keep the `map` variable private 🤔

Comment (alpadev) [reply js/src/dom/data.js]: If you keep the exported API, it should be still private as it's defined in module scope I guess.

Comment (rohit2sharma95) [reply js/src/dom/data.js]: Okay, let me remove IIFE then 🙂

Comment (alpadev) [reply js/src/dom/data.js]: How about this? 😉
```js
const mapData = new Map()

const Data = {
  setData(instance, data) {
    mapData.set(instance, data)
  },
  getData(instance) {
    return mapData.get(instance) || null
  },
  removeData(instance) {
    mapData.delete(instance)
  }
}

export default Data
```

**Comment by rohit2sharma95** [comment]
Now seems good? @alpadev 🙂

**Comment by alpadev** [comment]
About the missing unit tests. Do you think we need to add some more?

**Comment by rohit2sharma95** [comment]
Yes, because I just removed some unit tests (Just to pass tests) instead of updating them.
So definitely those tests should be added 🙂

**Comment by rohit2sharma95** [comment]
Who will do work on those tests 😄 
Just asking so that we both do not work for the same

**Comment by alpadev** [comment]
As you like.

**Comment by rohit2sharma95** [comment]
Okay I will do then 👍

**Comment by alpadev** [comment]
One of the tests you removed wasn't properly done anyway 😁

I think some of them are of no use anymore because you removed the key/bsKey but `'should remove data if something is stored'` could be readded. Thats like a complete workflow 👍

**Comment by rohit2sharma95** [comment]
Okay, then I think we should move this PR to the main repository first 😄

**Pull request merged**